strapi/strapi
# [WIP]chore: File block for Slate block editor

### What does it do?

It:
- Adds a File option to the blocks type selector.
- Adds a file block to render uploaded files on the content following the example:
![image](https://github.com/user-attachments/assets/1d5c948b-1d22-428b-8a8c-b6b4ea4a2d0a)
- Handles file uploading/choosing with a Dialog.
- 
### Why is it needed?

- Content Managers want to add not only images but files to their content.
- Will make it easy for frontends to render files on the websites.

### How to test it?
Create a new block, choose it to be a file block, choose the file from the Media Dialog, see it rendered on the content, and persist when refreshing the page.

### Related issue(s)/PR(s)
It is related to the following feature request: https://feedback.strapi.io/customization/p/add-audio-file-and-video-blocks-to-blocks-component.

### TODO
- [ ] Style the rendered file box on content.
- [ ] Finish the E2E test for file.

### Notes
- Right now, some unitary tests for the [Image block](https://github.com/strapi/strapi/blob/develop/packages/core/admin/admin/src/content-manager/components/BlocksInput/Blocks/tests/Image.test.tsx#L95) are broken, as we base our unitary test on there, we are unable to fix them now.
- As there is no E2E for the Image block, we need some help to test the Media Dialog on E2E tests.
- [Setting up Strapi is broken now](https://github.com/strapi/strapi/issues/20857), we will wait till it is fixed.
- [Media Dialog for Image Block was not working before](https://github.com/strapi/strapi/issues/20697), we need to have it fixed to test the File Block behavior and to finish the file box styling.

### PROJECT:
- [ ] File
- [ ] Audio
- [ ] Video
- [ ] Media Dialog (as the base for all of them, including the image, is a media file, we could have a MediaDialog for all blocks and just pass which files will be handled as a parameter)
- [ ] Character count
